app/views/common/_ganalytics.html.erb

```diff
@@ -1,18 +1,23 @@
 <% key ||= Conf.ga_key %>
 <% if Rails.env == "production" %>
-  <!-- google analytics -->
-  <script type="text/javascript">
-   var _gaq = _gaq || [];
-   _gaq.push(['_setAccount', '<%= key %>']);
-   _gaq.push(['_trackPageview']);
-
-   (function() {
-     var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-     ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
-     var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
-   })();
+  <!-- Global site tag (gtag.js) - Google Analytics -->
+  <script async src='https://www.googletagmanager.com/gtag/js?id=<%= key %>'></script>
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
+    gtag('js', new Date());
 
+    gtag('config', '<%= key %>');
   </script>
+
+  <!-- Google Tag Manager -->
+  <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+      new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+  })(window,document,'script','dataLayer','GTM-M7V3ZMB');</script>
```

Comment:
because this key is used it two places `GTM-M7V3ZMB`, could we put it in the `Conf` file so that it only needs to be set once?  I assume that key is related to our GA account, right?  If it's really not secret (which I agree cuz it's just javascript), we can put it in `config.yml` with all the other config keys (instead of in `config.keys.yml`

And then I wonder about the `key` which is the old GA key, right?  Is this the way we have one spot that provides analytics for all sites?

Reply:
That's a good point. I didn't see the GTM key being used at two different places. I will update that.

For GA, I think the tracking key never changed. But as I read more about it today, I think I made a mistake here replacing the old tracking code. The GA4(implemented by gtag.js here) needs to be set up **in parallel** with the Universal Analytics. the GA4 does NOT include historical data and offers features we may not need yet, like YouTube play data and downloads data. I will restore the old tracking code.